---
- name: Send API request and debug complete response
  hosts: localhost
  gather_facts: false
  vars:
    error_source: ""
    error_logs: ""
    ask_content: 発生しているエラーの内容と考えられる原因、調査方法を日本語で教えて下さい。
  tasks:
    - name: Set facts
      ansible.builtin.set_fact:
        rhelai_server: "{{ rhelai_server_input | default('rhelai') }}"
        rhelai_port: "{{ rhelai_port_input | default('8000') }}"
        rhelai_token: "{{ rhelai_token_input | default('ansible123!') }}"
        gpt_prompt: |
          これは {{ error_source }} です。
          {{ ask_content }}

          {{ error_logs }}

    - name: Send POST request using uri module
      ansible.builtin.uri:
        url: "http://{{ rhelai_server }}:{{ rhelai_port }}/v1/completions"
        method: POST
        headers:
          accept: "application/json"
          Content-Type: "application/json"
          Authorization: "Bearer {{ rhelai_token }}"
        body:
          prompt: "{{ gpt_prompt | default('What is the capital of the USA?') }}"
          max_tokens: "{{ max_tokens | default('50') }}"
          model: "/root/.cache/instructlab/models/granite-8b-lab-v1"
          temperature: "{{ input_temperature | default(0) }}"
          top_p: "{{ input_top_p | default(1) }}"
          n: "{{ input_n | default(1) }}"
        body_format: json
        return_content: true
        status_code: 200
        timeout: 60
      register: gpt_response

    - name: Check if response is JSON and debug output
      ansible.builtin.debug:
        msg: "{{ gpt_response.json if gpt_response.json is defined else gpt_response.content }}"

    - name: Set stats for next controller job
      ansible.builtin.set_stats:
        data:
          gpt_response: "{{ gpt_response }}"

    - name: Debug gpt response
      ansible.builtin.debug:
        var: gpt_response
